yarn-project/aztec-node/src/sentinel/sentinel.ts

```diff
@@ -84,7 +97,90 @@ export class Sentinel implements L2BlockStreamEventHandler {
           this.slotNumberToBlock.delete(key);
         }
       }
+    } else if (event.type === 'chain-proven') {
+      await this.handleChainProven(event);
+    }
+  }
+
+  protected async handleChainProven(event: L2BlockStreamEvent) {
+    if (event.type !== 'chain-proven') {
+      return;
+    }
+    const blockNumber = event.block.number;
+    const block = await this.archiver.getBlock(blockNumber);
+    if (!block) {
+      this.logger.error(`Failed to get block ${blockNumber}`, { block });
+      return;
     }
+
+    const epoch = getEpochAtSlot(block.header.getSlot(), await this.archiver.getL1Constants());
+    this.logger.info(`Computing proven performance for epoch ${epoch}`);
+    const performance = await this.computeProvenPerformance(epoch);
+    this.logger.info(`Proven performance for epoch ${epoch}`, performance);
+
+    await this.updateProvenPerformance(epoch, performance);
+    this.handleProvenPerformance(performance);
+  }
+
+  protected async computeProvenPerformance(epoch: bigint) {
+    const headers = await this.archiver.getBlockHeadersForEpoch(epoch);
+    const provenSlots = headers.map(h => h.getSlot());
+    const fromSlot = provenSlots[0];
```

Comment:
note to self, check that chain proven chain cannot progress if there are no blocks in an epoch. (if so this may throw)